definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/test_connect.js

```diff
@@ -0,0 +1,495 @@
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passthrough={123} forMapStateToProps={321}/>;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passthrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testWithStatelessFunctionalComponent() {
+  type Props = {passthrough: number, fromStateToProps: string};
+  const Com = (props: Props) => <div>{props.passthrough} {props.fromStateToProps}</div>
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for passthrough
+  <Connected passthrough={''} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passthrough={123} forMapStateToProps={321} />;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passthrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testMapStateToPropsDoesNotNeedProps() {
+  type Props = {passthrough: number, fromStateToProps: string};
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough}</div>;
+    }
+  }
+
+  type State = {a: string}
+  const mapStateToProps = (state: State) => {
+    return {
+      fromStateToProps: state.a
+    }
+  }
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123}/>;
+  //$ExpectError component property passthrough not found
+  <Connected />;
+}
+
+function testMapDispatchToProps() {
+  type Props = {
+    passthrough: number,
+    fromMapDispatchToProps: string,
+    fromMapStateToProps: string
+  };
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>
+        {this.props.passthrough}
+        {this.props.fromMapDispatchToProps}
+        {this.props.fromMapStateToProps}
+        </div>;
+    }
+  }
+
+  type State = {a: number}
+  type MapStateToPropsProps = {forMapStateToProps: string}
+  const mapStateToProps = (state: State, props: MapStateToPropsProps) => {
+    return {
+      fromMapStateToProps: 'str' + state.a
+    }
+  }
+  type MapDispatchToPropsProps = {forMapDispatchToProps: string}
+  const mapDispatchToProps = (dispatch: *, ownProps: MapDispatchToPropsProps) => {
+    return {fromMapDispatchToProps: ownProps.forMapDispatchToProps}
+  }
+  const Connected = connect(mapStateToProps, mapDispatchToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
+  //$ExpectError passthrough missing
```

Comment:
removing this causes

```
    Error -------------------------------------------------------------------------------------------- test_connect.js:260:3
    
    Cannot create `Connected` element because property `fromMapDispatchToProps` is missing in props [1] but exists in
    `Props` [2].
    
       test_connect.js:260:3
       260|   <Connected forMapStateToProps={'data'} forMapDispatchToProps={'more data'} />;
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ [1]
    
    References:
       test_connect.js:236:37
       236|   class Com extends React.Component<Props> {
                                                ^^^^^ [2]
```

instead of `passthrough` missing.